### PR TITLE
lmod: update to 8.7.53

### PR DESCRIPTION
**Pull request by martin-g**
Touch python-scipy.spec to test lmod

### CODE REVIEW

**Comment by martin-g** [comment]
Related to #2034

**Comment by martin-g** [comment]
`. /etc/profile.d/lmod.sh` is sourced without problems but I think this is still the old/working version.
The new version is installed:
```
Upgrading:
2024-10-17T12:31:48.8475741Z  lmod-ohpc                              x86_64  8.7.53-9999.ci.ohpc.1 @commandline          251 k
```
but its `/etc/profile.d/lmod.sh` is not sourced, so I cannot say whether it works or not...

File: components/dev-tools/scipy/SPECS/python-scipy.spec [hunk 1]
```diff
@@ -26,7 +26,7 @@ Requires: openblas-%{compiler_family}%{PROJ_DELIM}
 
 Name:           %{python_prefix}-%{pname}-%{compiler_family}-%{mpi_family}%{PROJ_DELIM}
 Version:        1.5.4
-Release:        1%{?dist}
+Release:        2%{?dist}
```
Comment (martin-g) [inline components/dev-tools/scipy/SPECS/python-scipy.spec]: This change is just to trigger the testing for another (non-lmod) module. It should be reverted before merge.

**Comment by adrianreber** [comment]
You can also download the resulting RPM and try it locally in a container. It is attached to the GitHub Actions page for 24 hours.

**Comment by martin-g** [comment]
> You can also download the resulting RPM and try it locally in a container. It is attached to the GitHub Actions page for 24 hours.

```
root@431886de7e52 /# dnf install -y lmod-ohpc
OpenHPC-3 - Base                                                                                                                                                                     884 kB/s | 2.2 MB     00:02    
OpenHPC-3 - Updates                                                                                                                                                                  699 kB/s | 2.1 MB     00:03    
Last metadata expiration check: 0:00:01 ago on Thu Oct 17 13:00:00 2024.
Dependencies resolved.
=====================================================================================================================================================================================================================
 Package                                             Architecture                                Version                                                  Repository                                            Size
=====================================================================================================================================================================================================================
Installing:
 lmod-ohpc                                           x86_64                                      8.7.37-310.ohpc.4.1                                      OpenHPC-updates                                      267 k
Installing dependencies:
 lua-filesystem                                      x86_64                                      1.8.0-1.oe2203sp3                                        everything                                            17 k
 lua-posix                                           x86_64                                      35.0-1.oe2203sp3                                         OS                                                    99 k
 tcsh                                                x86_64                                      6.23.00-3.oe2203sp3                                      OS                                                   354 k

Transaction Summary
=====================================================================================================================================================================================================================
Install  4 Packages

Total download size: 737 k
Installed size: 3.1 M
Downloading Packages:
(1/4): lua-posix-35.0-1.oe2203sp3.x86_64.rpm                                                                                                                                         231 kB/s |  99 kB     00:00    
(2/4): tcsh-6.23.00-3.oe2203sp3.x86_64.rpm                                                                                                                                           779 kB/s | 354 kB     00:00    
(3/4): lua-filesystem-1.8.0-1.oe2203sp3.x86_64.rpm                                                                                                                                    51 kB/s |  17 kB     00:00    
(4/4): lmod-ohpc-8.7.37-310.ohpc.4.1.x86_64.rpm                                                                                                                                      229 kB/s | 267 kB     00:01    
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Total                                                                                                                                                                                203 kB/s | 737 kB     00:03     
OpenHPC-3 - Updates                                                                                                                                                                  1.3 MB/s | 1.4 kB     00:00    
Importing GPG key 0x40A90CC8:
 Userid     : "OpenHPC3 OBS Project <OpenHPC3@private>"
 Fingerprint: 6E19 BE11 D9A0 82C0 0A7D B41A 7AC0 5F09 40A9 0CC8
 From       : /etc/pki/rpm-gpg/RPM-GPG-KEY-OpenHPC-3
Key imported successfully
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                                                                                                                                                             1/1 
  Installing       : lua-filesystem-1.8.0-1.oe2203sp3.x86_64                                                                                                                                                     1/4 
  Running scriptlet: tcsh-6.23.00-3.oe2203sp3.x86_64                                                                                                                                                             2/4 
  Installing       : tcsh-6.23.00-3.oe2203sp3.x86_64                                                                                                                                                             2/4 
  Running scriptlet: tcsh-6.23.00-3.oe2203sp3.x86_64                                                                                                                                                             2/4 
  Installing       : lua-posix-35.0-1.oe2203sp3.x86_64                                                                                                                                                           3/4 
  Installing       : lmod-ohpc-8.7.37-310.ohpc.4.1.x86_64                                                                                                                                                        4/4 
  Running scriptlet: lmod-ohpc-8.7.37-310.ohpc.4.1.x86_64                                                                                                                                                        4/4 
  Verifying        : lmod-ohpc-8.7.37-310.ohpc.4.1.x86_64                                                                                                                                                        1/4 
  Verifying        : lua-posix-35.0-1.oe2203sp3.x86_64                                                                                                                                                           2/4 
  Verifying        : tcsh-6.23.00-3.oe2203sp3.x86_64                                                                                                                                                             3/4 
  Verifying        : lua-filesystem-1.8.0-1.oe2203sp3.x86_64                                                                                                                                                     4/4 

Installed:
  lmod-ohpc-8.7.37-310.ohpc.4.1.x86_64                  lua-filesystem-1.8.0-1.oe2203sp3.x86_64                  lua-posix-35.0-1.oe2203sp3.x86_64                  tcsh-6.23.00-3.oe2203sp3.x86_64                 

Complete!

[root@431886de7e52 /]#  . /etc/profile.d/lmod.sh

[root@431886de7e52 /]# dnf install /oe/home/ohpc/rpmbuild/RPMS/x86_64/lmod-ohpc-8.7.53-9999.ci.ohpc.1.x86_64.rpm 
Last metadata expiration check: 0:00:52 ago on Thu Oct 17 13:00:00 2024.
Dependencies resolved.
=====================================================================================================================================================================================================================
 Package                                         Architecture                                 Version                                                       Repository                                          Size
=====================================================================================================================================================================================================================
Upgrading:
 lmod-ohpc                                       x86_64                                       8.7.53-9999.ci.ohpc.1                                         @commandline                                       280 k

Transaction Summary
=====================================================================================================================================================================================================================
Upgrade  1 Package

Total size: 280 k
Is this ok [y/N]: y
Downloading Packages:
Running transaction check
Transaction check succeeded.
Running transaction test
Transaction test succeeded.
Running transaction
  Preparing        :                                                                                                                                                                                             1/1 
  Upgrading        : lmod-ohpc-8.7.53-9999.ci.ohpc.1.x86_64                                                                                                                                                      1/2 
  Cleanup          : lmod-ohpc-8.7.37-310.ohpc.4.1.x86_64                                                                                                                                                        2/2 
  Verifying        : lmod-ohpc-8.7.53-9999.ci.ohpc.1.x86_64                                                                                                                                                      1/2 
  Verifying        : lmod-ohpc-8.7.37-310.ohpc.4.1.x86_64                                                                                                                                                        2/2 

Upgraded:
  lmod-ohpc-8.7.53-9999.ci.ohpc.1.x86_64                                                                                                                                                                             

Complete!
[root@431886de7e52 /]#  . /etc/profile.d/lmod.sh
[root@431886de7e52 /]# 

```

The new version seems OK!

I my last commit I added a test case for lmod. Hopefully it will pass too!

**Comment by martin-g** [comment]
@adrianreber I think this PR is ready for review!

File: tests/admin/lmod/LICENSE [hunk 2]
```diff
@@ -0,0 +1,24 @@
+Copyright (c) 2015, Intel Corporation
```
Comment (adrianreber) [inline tests/admin/lmod/LICENSE]: This license file seems unnecessary.

Comment (martin-g) [reply tests/admin/lmod/LICENSE]: Removed!
Copy/paste from the spack/ folder

**Pull request merged**